lib/constants.dart

```diff
@@ -1,4 +1,50 @@
 class Constants {
   static const packageName = 'amplitude-flutter';
   static const packageVersion = '3.16.1';
+
+  // Configuration options
+  static const flushQueueSize = 30;
+  static const flushIntervalMillis = 30 * 1000; // 30 seconds
+  static const identifyBatchIntervalMillis = 30 * 1000; // 30 seconds
+  static const defaultInstanceName = '\$default_instance';
+  static const logLevel = 'info';
+  static const flushMaxRetries = 5;
+  static const minTimeBetweenSessionsMillis = 5 * 60 * 1000; // 5 minutes
+
+  // Tracking options available on iOS, Android, and Web
+  static const String ampTrackingOptionIpAddress = "ip_address";
+  static const String ampTrackingOptionLanguage = "language";
+  static const String ampTrackingOptionPlatform = "platform";
+  // Tracking options only available on iOS, Android
+  static const String ampTrackingOptionCarrier = "carrier";
+  static const String ampTrackingOptionCity = "city";
+  static const String ampTrackingOptionCountry = "country";
+  static const String ampTrackingOptionDeviceModel = "device_model";
+  static const String ampTrackingOptionDeviceManufacturer = "device_manufacturer";
+  static const String ampTrackingOptionDma = "dma";
+  static const String ampTrackingOptionOsName = "os_name";
+  static const String ampTrackingOptionOsVersion = "os_version";
+  static const String ampTrackingOptionRegion = "region";
+  static const String ampTrackingOptionVersionName = "version_name";
+  // Tracking options only available on iOS
+  static const String ampTrackingOptionIdfv = "idfv";
+  // Tracking options only available on Android
+  static const String ampTrackingOptionAdid = 'adid';
+  static const String ampTrackingOptionAppSetId = 'app_set_id';
+  static const String ampTrackingOptionDeviceBrand = "device_brand";
+  static const String ampTrackingOptionLatLag = "Lat_Lng";
+  static const String ampTrackingOptionApiLevel = "api_level";
 }
+
+enum LogLevel {
+  off,
+  error,
+  warn,
+  log,
+  debug,
+}
+
+enum ServerZone {
+  US,
+  EU,
+}
```

Comment:
nit: not sure if there is a linter in dart, normally there is an empty line at the end of file.

Reply:
I just set up flutter linter. No empty line at the end of file doesn't break lint rules. 

Here is the result. All of them are from existing code. I will fix them later. Create a [ticket](https://amplitude.atlassian.net/browse/AMP-93080) for setting up ci.
```
flutter analyze                                                                              ─╯
Analyzing Amplitude-Flutter...

   info • Unnecessary 'null' comparison before assigning a value • lib/amplitude.dart:18:5 •
          prefer_conditional_assignment
   info • Unnecessary 'new' keyword • lib/amplitude.dart:23:42 • unnecessary_new
   info • Unnecessary use of '??' with 'null' • lib/amplitude_web.dart:32:24 •
          unnecessary_null_in_if_null_operators
   info • Unnecessary use of '??' with 'null' • lib/amplitude_web.dart:54:28 •
          unnecessary_null_in_if_null_operators
   info • Unnecessary use of '??' with 'null' • lib/amplitude_web.dart:101:35 •
          unnecessary_null_in_if_null_operators
   info • Unnecessary 'new' keyword • lib/amplitude_web.dart:205:25 • unnecessary_new
   info • The constant name 'US' isn't a lowerCamelCase identifier • lib/constants.dart:48:3 •
          constant_identifier_names
   info • The constant name 'EU' isn't a lowerCamelCase identifier • lib/constants.dart:49:3 •
          constant_identifier_names
   info • The constant name 'OP_SET' isn't a lowerCamelCase identifier • lib/identify.dart:6:23 •
          constant_identifier_names
   info • The constant name 'OP_SET_ONCE' isn't a lowerCamelCase identifier •
          lib/identify.dart:7:23 • constant_identifier_names
   info • The constant name 'OP_ADD' isn't a lowerCamelCase identifier • lib/identify.dart:8:23 •
          constant_identifier_names
   info • The constant name 'OP_APPEND' isn't a lowerCamelCase identifier • lib/identify.dart:9:23
          • constant_identifier_names
   info • The constant name 'OP_UNSET' isn't a lowerCamelCase identifier • lib/identify.dart:10:23
          • constant_identifier_names
   info • The constant name 'OP_PREPEND' isn't a lowerCamelCase identifier •
          lib/identify.dart:11:23 • constant_identifier_names
   info • The constant name 'OP_PREINSERT' isn't a lowerCamelCase identifier •
          lib/identify.dart:12:23 • constant_identifier_names
   info • The constant name 'OP_POSTINSERT' isn't a lowerCamelCase identifier •
          lib/identify.dart:13:23 • constant_identifier_names
   info • The constant name 'OP_REMOVE' isn't a lowerCamelCase identifier • lib/identify.dart:14:23
          • constant_identifier_names
   info • The constant name 'OP_CLEAR_ALL' isn't a lowerCamelCase identifier •
          lib/identify.dart:15:23 • constant_identifier_names
   info • Use 'const' with the constructor to improve performance • lib/playground.dart:10:28 •
          prefer_const_constructors
   info • Use 'const' with the constructor to improve performance • lib/playground.dart:14:24 •
          prefer_const_constructors
   info • Use 'const' with the constructor to improve performance • lib/playground.dart:17:21 •
          prefer_const_constructors
   info • Use 'const' with the constructor to improve performance • lib/playground.dart:20:20 •
          prefer_const_constructors
   info • Use 'const' with the constructor to improve performance • lib/playground.dart:22:36 •
          prefer_const_constructors
   info • The parameter name 'int' matches a visible type name • lib/web/amplitude_js.dart:19:41 •
          avoid_types_as_parameter_names

24 issues found. (ran in 1.1s)
```